target/linux/mpc85xx/image/p1020.mk

```diff
@@ -39,3 +39,17 @@ define Device/ocedo_panda
 endef
 TARGET_DEVICES += ocedo_panda
 
+define Device/watchguard_t30-w
+  DEVICE_VENDOR := WatchGuard
+  DEVICE_MODEL := T30-W
+  DEVICE_PACKAGES := kmod-ath10k-ct ath10k-firmware-qca988x-ct \
+    kmod-usb2 kmod-usb-storage block-mount kmod-fs-vfat kmod-nls-cp437 \
+    kmod-nls-iso8859-1 kmod-loop losetup kmod-fs-ext4 e2fsprogs tune2fs
+  KERNEL_SIZE := 8192k
+  KERNEL = kernel-bin | gzip | fit gzip $(KDIR)/image-$$(DEVICE_DTS).dtb
+  IMAGES := fdt.bin sysupgrade.bin sysupgrade-dtb.bin
+  IMAGE/fdt.bin := append-dtb
+  IMAGE/sysupgrade.bin := sysupgrade-dtb-tar \
```

Comment:
If that one single ext2 has to stick around, maybe you could do the integration like x86 does and integrate the dtb right there. You could also make a something like the rpi does with their sd-card factory images for the (prepartitioned?) sd-card.

Reply:
uboot reads the kernel and dtb from the ext2 partition on partition 3 on the MMC (sdcard).
sysupgrade will need to write the new kernel and new dtb into that partition each time, so we will need a special sysupgrade script no matter what.

Is there an issue with the way that I have handled sysupgrades ?